src/useFormStates.ts

```diff
@@ -44,20 +57,22 @@ export function useFormStates<T, I>(opts: UseFormStatesOpts<T, I>): { getObjectS
   return {
     getObjectState: (input) => {
       const existing = objectStateCache[getId(input)];
-      const form: ObjectState<T> = existing
-        ? existing[0]
-        : createObjectState(config, initValue(config, map ? { map, input } : input), {
```

Comment:
I expanded this ternary so that we can call `addRules` conditionally after it's created.

...actually maybe I didn't need to do this, vs. just adding `addRules` to the `if (!existing)` block...but this also seems slightly cleaner now.